src/pages/index.js

```diff
@@ -92,26 +149,98 @@ function fillProfileForm() {
 
 profileEditButton.addEventListener("click", () => {
   formValidators["profile-form"].resetValidation();
+  popupWithForms["edit-modal"].resetButtonText();
   popupWithForms["edit-modal"].open();
   fillProfileForm();
 });
 
+profileAvatar.addEventListener("mouseover", () => {
+  avatarEditButton.classList.add("profile__avatar-button_visible");
+});
+
+avatarEditButton.addEventListener("click", () => {
+  //reset the validation before opening it
+  formValidators["avatar-form"].resetValidation();
+  popupWithForms["avatar-modal"].resetButtonText();
+  popupWithForms["avatar-modal"].open();
+});
+
+const updateProfilePicture = (inputValues) => {
+  api
+    .updateProfilePicture(inputValues["avatar-link"])
+    .then((res) => {
+      profileInfo.updateUserAvatar(res["avatar"]);
+      popupWithForms["avatar-modal"].changeButtonText();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  //retrieveProfileInfo();
+  /*api
+    .loadUserInfo()
+    .then((profileData) => {
+      popupWithForms["avatar-modal"].changeButtonText();
+      profileInfo.updateUserAvatar(profileData["avatar"]);
+    })
+    .catch((err) => {
+      console.error(err);
+    });*/
+  popupWithForms["avatar-modal"].close();
+};
+
 function updateProfileModal(inputValues) {
   profileInfo.setUserInfo(inputValues["Name"], inputValues["About me"]);
+  api
+    .editUserInfo(inputValues["Name"], inputValues["About me"])
+    .then((res) => {
+      popupWithForms["edit-modal"].changeButtonText();
```

Comment:
You need to start the loading effect **before** you make a request to see the loading...